### PR TITLE
A few more math libraries

### PR DESCRIPTION
**Pull request by catap**
#### Description

All of this small library is dependency for Sage on port of which I'm working.

###### Type(s)

- [x] enhancement

###### Tested on

macOS 12.6.3 21G419 x86_64
Xcode 14.2 14C18

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by catap** [comment]
@mascguy may I ask you to review this PR? I'm still working on bring `port install sage` here, that requires some additional work.

File: math/BRiAl/Portfile [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem              1.0
+PortGroup               github 1.0
+PortGroup               boost 1.0
+
+github.setup            BRiAl BRiAl 1.2.12
```
Comment (mascguy) [inline math/BRiAl/Portfile:7]: Add a `revision 0`

File: math/m4ri/Portfile [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem              1.0
+PortGroup               bitbucket 1.0
+
+bitbucket.setup         malb m4ri 20200125 release-
```
Comment (mascguy) [inline math/m4ri/Portfile:6]: Add a `revision 0`

File: math/palp/Portfile [hunk 3]
```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem              1.0
+
+name                    palp
+version                 2.20
```
Comment (mascguy) [inline math/palp/Portfile:6]: Add a `revision 0`

**Comment by mascguy** [comment]
LGTM, other than adding `revision` to your portfiles.

**Comment by catap** [comment]
Here it is

**Review by mascguy — APPROVED** [review]

**Pull request merged**